internal/pkg/cfa/function.go

```diff
@@ -0,0 +1,124 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package cfa implements cross-function analysis. The analyzer
+// defined in this package looks at every function in the transitive
+// dependencies of the program being analyzed and creates an abstraction
+// for each one that can be used to determine what the function does with
+// each of its arguments.
+
+package cfa
+
+import (
+	"fmt"
+	"sort"
+	"strconv"
+	"strings"
+
+	"golang.org/x/tools/go/ssa"
+)
+
+// A Function is an abstraction for a Go function.
+// It can be queried about what it does with its arguments.
+type Function interface {
+	// Does this function's nth argument reach a sink?
+	Sinks(arg int) bool
+
+	// If this argument is tainted, which return values are tainted?
+	Taints(arg int) []int
+}
+
+type sink struct{}
```

Comment:
Are these structs meant to represent how we currently handle sinks and taint?  Are they meant for consumption outside this package as we iterate towards integrating this interface into existing code?  Should they be exported?

A comment would be good.  Otherwise, it's appears to be dead code.

Reply:
`sink` and `sanitizer` are only intended for in-package use. Outside users will be expected to interact with them through the interface. Should I add comments even though they're unexported?

`genericFunc` should be exported though, since it has an additional method outside the interface that users will need. I've exported it and added a comment.

Reply:
I've unexported `genericFunc` due to no longer needing the extra out-of-interface method. I've kept the comment though.